### ISSUE TITLE
[SideNav] item styling/indentation is invalid

### ISSUE THREAD

**Issue by davidicus** [carbon-design-system/carbon-addons-iot-react#1584]
### What package is this for?

- [x] React
- [ ] Angular

### Describe the bug

Regression of SideNav styles with latest Carbon update. 

![image](https://user-images.githubusercontent.com/1590966/92538527-ea245380-f20c-11ea-965a-caafd23b5dbf.png)

**Comment by tay1orjones** [comment]
Spec on the design site https://pages.github.ibm.com/ai-applications/design/components/ui-shell-side-nav/usage

**Comment by tay1orjones** [comment]
Indentation is off compared to Carbon
![image](https://user-images.githubusercontent.com/3360588/93940525-3930b500-fcf2-11ea-99c7-32f0d3110e53.png)

**Comment by tay1orjones** [comment]
There's more details reported in https://github.com/carbon-design-system/carbon-addons-iot-react/issues/1618

> The text in the SideNav items shouldn’t be bold (it should also be white, not sure why in the React implementation some items appear with grey text).

**Comment by tay1orjones** [comment]
The deploy preview for the PR that was the last one to be merged before the Carbon upgrade shows what it should look like https://deploy-preview-1552--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-sidenav--sidenav-component

**Comment by dianatran18** [comment]
Hey @davidicus not sure if this is the right issue but here are the updated specs for the nav.

I tried to map the color tokens to Gray 10 theme colors as much as I could, but there's one color that doesn't map to anything (I don't think?). It's the same color Carbon uses in their header for their hover states on icon buttons #2C2C2C.

The one thing I'm not sure about is whether the selected text would be semi-bold, or stay regular. It looks like the old design spec didn't have semi-bold text, but in the code it did. I kept it as regular to match the old design spec, but if something changed along the way that I'm not aware about let me know and I can update it! 

![IoT-sidenav-specs](https://user-images.githubusercontent.com/43549567/96212723-7b41c500-0f45-11eb-85ff-046ecbe41834.png)
